src/actions/posts.js

```diff
@@ -759,6 +760,28 @@ export function getPostsAround(channelId, postId, perPage = Posts.POST_CHUNK_SIZ
     };
 }
 
+// getThreadsForPosts is intended for an array of posts that have been batched
+// (see the actions/websocket_actions/handleNewPostEvents function in the webapp)
+export function getThreadsForPosts(posts, dispatch, getState) {
+    if (!Array.isArray(posts) || posts.length === 0) {
+        return Promise.resolve();
+    }
+
+    const promises = [];
+
+    posts.forEach((post) => {
+        if (!post.root_id) {
+            return;
+        }
+        const rootPost = Selectors.getPost(getState(), post.root_id);
+        if (post.root_id && !rootPost) {
```

Comment:
nit: we've already checked `post.root_id` above:
```suggestion
        if (!rootPost) {
```

Reply:
d'oh! Thanks. :)